pkg/cmd/codespace/codespace_selector.go

```diff
@@ -0,0 +1,123 @@
+package codespace
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/v2/internal/codespaces/api"
+	"github.com/spf13/cobra"
+)
+
+type CodespaceSelector struct {
```

Comment:
I think this turned out great! Real clean implementation! I think in the future it would be cool to add in some more advanced filtering ability on properties of the codespace, such as selecting only running codespaces, so that the selector can be used in more of those intricate commands like delete.

Reply:
@samcoe 💯 I had that exact same thought when I was forced to not use the selector directly for the `stop` command! More fun for the future 😄